crates/subspace-networking/src/node_runner.rs

```diff
@@ -211,6 +217,31 @@ where
         let local_peer_id = *self.swarm.local_peer_id();
         let connected_peers = self.swarm.connected_peers().cloned().collect::<Vec<_>>();
 
+        // Handle DSN connection signaling
+        // TODO: consider moving it a separate watcher with decoupled check period.
+        let current_dsn_connection_state = !connected_peers.is_empty();
```

Comment:
Why do you check it during dialing and not connection/disconnection?
Dialing happens A LOT, connection/disconnection much less frequently and that is when the state *actually* changes. By the time you get to dialing you can already be offline, but the value isn't updated yet.